rotkehlchen/chain/ethereum/modules/curve/decoder.py

```diff
@@ -48,28 +53,208 @@
 # Deposit contracts are retrieved from the links below Deposit<pool>:
 # https://curve.readthedocs.io/ref-addresses.html#base-pools
 # https://curve.readthedocs.io/ref-addresses.html#metapools
+# a mapping of deposit contracts to their coins & pool token.
 CURVE_DEPOSIT_CONTRACTS = {
-    string_to_evm_address('0x094d12e5b541784701FD8d65F11fc0598FBC6332'),  # curve usdn
-    string_to_evm_address('0xB0a0716841F2Fc03fbA72A891B8Bb13584F52F2d'),  # curve ust
-    string_to_evm_address('0x3c8cAee4E09296800f8D29A68Fa3837e2dae4940'),  # curve usdp
-    string_to_evm_address('0xF1f85a74AD6c64315F85af52d3d46bF715236ADc'),  # curve usdk
-    string_to_evm_address('0xBE175115BF33E12348ff77CcfEE4726866A0Fbd5'),  # curve rsv
-    string_to_evm_address('0x803A2B40c5a9BB2B86DD630B274Fa2A9202874C2'),  # curve musd
-    string_to_evm_address('0x1de7f0866e2c4adAC7b457c58Cc25c8688CDa1f2'),  # curve linkusd
-    string_to_evm_address('0x09672362833d8f703D5395ef3252D4Bfa51c15ca'),  # curve husd
-    string_to_evm_address('0x64448B78561690B70E17CBE8029a3e5c1bB7136e'),  # curve gusd
-    string_to_evm_address('0x61E10659fe3aa93d036d099405224E4Ac24996d0'),  # curve dusd
-    string_to_evm_address('0xbBC81d23Ea2c3ec7e56D39296F0cbB648873a5d3'),  # curve y
-    string_to_evm_address('0xb6c057591E073249F2D9D88Ba59a46CFC9B59EdB'),  # curve busd
-    string_to_evm_address('0xeB21209ae4C2c9FF2a86ACA31E123764A3B6Bc06'),  # curve compound
-    string_to_evm_address('0xA50cCc70b6a011CffDdf45057E39679379187287'),  # curve pax
-    string_to_evm_address('0xFCBa3E75865d2d561BE8D220616520c171F12851'),  # curve susd v2
-    string_to_evm_address('0x331aF2E331bd619DefAa5DAc6c038f53FCF9F785'),  # curve zap
-    string_to_evm_address('0xac795D2c97e60DF6a99ff1c814727302fD747a80'),  # curve usdt
-    string_to_evm_address('0x071c661B4DeefB59E2a3DdB20Db036821eeE8F4b'),  # curve bbtc
-    string_to_evm_address('0xd5BCf53e2C81e1991570f33Fa881c49EEa570C8D'),  # curve obtc
-    string_to_evm_address('0x11F419AdAbbFF8d595E7d5b223eee3863Bb3902C'),  # curve pbtc
-    string_to_evm_address('0xaa82ca713D94bBA7A89CEAB55314F9EfFEdDc78c'),  # curve tbtc
+    string_to_evm_address('0x094d12e5b541784701FD8d65F11fc0598FBC6332'): (  # curve usdn
+        {
+            string_to_evm_address('0x6B175474E89094C44Da98b954EedeAC495271d0F'),  # dai
```

Comment:
I think it should be sufficient to map only to lp token / address of the pool. And coins can be found in the curve cache, can't they?

Reply:
Some deposit zap contracts don't have a pool/base_pool so the curve_pool cache does not work here.

e.g. `0xA50cCc70b6a011CffDdf45057E39679379187287` `0xbBC81d23Ea2c3ec7e56D39296F0cbB648873a5d3` `0xb6c057591E073249F2D9D88Ba59a46CFC9B59EdB` to name a few

Reply:
How can it be like that? Are you sure that these really curve deposit contracts if they don't interact with curve pools?

Btw, I checked and it seems like they do. Examples:
1. `0xA50cCc70b6a011CffDdf45057E39679379187287`: https://etherscan.io/tx/0x53ef827562d9283943f688e97ace1f5816ee12b3cd9c9be73023360e22c901ba#eventlog
2. `0xbBC81d23Ea2c3ec7e56D39296F0cbB648873a5d3`: https://etherscan.io/tx/0xd8eefcd63ea90ff98245a9a9ac5177b8b15a49e9270e8bfa9e832adb3757aab0#eventlog
3. `0xb6c057591E073249F2D9D88Ba59a46CFC9B59EdB`: https://etherscan.io/tx/0x025e996b1b77dede53436e4c97bcf098a47d8699e7ee1e644f3b33bd14d17f6b#eventlog

They all 3 interact with curve pools, or do I misunderstand something?

Reply:
The logs only show transfers of **coins**. No interaction with pools. Neither does the contract define them.